### ISSUE TITLE
Update task to resign add-ons without a COSE signature as part of SHA-1 deprecation

### ISSUE THREAD

**Issue by diox** [mozilla/addons-server#21996]
To prepare for Firefox deprecating (phase 1) and eventually (phase 2, not in scope for now) removing support for old PKCS#1 SHA-1 signatures in add-ons, we need to resign the limited number of add-ons that don't have a COSE signature so that they don't break.

We already have a task for this, `sign_addons`, that can be triggered with the `process_addons --task=resign_addons_for_cose` management command. This task was written years ago and never executed as the deprecation was pushed back.

We need to check that the existing code still makes sense and make a number of ajustements to this task, or at least double-check if it behaves like we want, notably regarding:
- [x] Which add-ons it's considering. At the moment it's processing all publicly available add-ons on AMO whose latest public listed version was created before April 5th 2019
  - [x] Do we want to deal with deleted add-ons ? Add-ons without a publicly available listed version currently ?
- [x] Which versions it's considering. At the moment it's re-signing and bumping the latest public listed version of each add-on it's processing.
- [x] How it bumps versions. At the moment it's doing something a bit unexpected, bumping the version number _on the existing version_ and overwriting the xpi. We likely don't want this, and want to create entirely new versions instead.
- [x] Email copy (needs to be rewritten entirely)
- [x] ~~Bonus: add a one-liner release notes to the version to explain what it is~~ (too annoying with locales)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/ADDSRV-754)

**Comment by diox** [comment]
We [agreed](https://mozilla-hub.atlassian.net/browse/AMO-209) to resign the current (last public listed) version of every affected non-dictionary add-ons that are public/not disabled by their developers. The query looks like this:
```sql
SELECT a.guid AS addon_id,
       a.average_daily_users
FROM addons a
JOIN versions v ON a.current_version = v.id
WHERE v.created <= "2019-01-23"
  AND a.addontype_id <> 3
  AND a.inactive = 0
  AND a.status = 4
```

**Comment by diox** [comment]
Interestingly, we ran a massive [repack for themes in May 2019](https://github.com/mozilla/addons-server/issues/10627), which explains why there are so few of them (~500 if I fix the threshold date to be April 5th) in the query above.

**Comment by diox** [comment]
QA: The resign is a command we need to coordinate together to run on dev & stage. You'll need to prepare some add-ons that should or should not be affected, and we'll have to manually alter their date to be be in the past to match the criteria.

Should be affected: add-ons that are public, non-invisible, extensions or themes only, with a current publicly listed version submitted before 2019-04-05.

**Comment by diox** [comment]
There were various failures running this on dev. Some due to the weird state some old add-ons are in, some more complicated, investigating.

**Comment by diox** [comment]
The vast majority of affected add-ons seem to bumped correctly after https://github.com/mozilla/addons-server/pull/22144 but we have a few outliers:

- Manifest ID mismatch  https://mozilla.sentry.io/issues/5199300487/?project=6224627
    - ID was likely manually altered in database
- Unexpected Error https://mozilla.sentry.io/issues/5199308651/?project=6224627
    - Add-on has no valid author and therefore we attempt to pass `None` as the user
- Trademark violation https://mozilla.sentry.io/issues/5198386651/?project=6224627
    - Needs to be investigated. Might be old add-ons (versions) created before we introduced
      trademark rules, or their authors lost permissions they used to have, so the
      existing versions can't directly be republished by the same user
- JSON error https://mozilla.sentry.io/issues/5198397100/?project=6224627
    - Broken version that shouldn't be public
- Add-on type doesn't match https://mozilla.sentry.io/issues/5199533625/?project=6224627
    - Add-on type was likely manually altered in database
- Missing compatibility https://mozilla.sentry.io/issues/5201311314/?project=6224627
    - Needs to be investigated.

**Comment by diox** [comment]
Another issue spotted by QA: theme previews were not generated. This is because we wrap each version creation in a transaction, but we fired the task to generate the previews as part of the creation before the transaction has had a chance to be committed.

It doesn't happen outside of this script because of https://github.com/mozilla/django-post-request-task/. For other scripts that need to do something like that we delay triggering the tasks in the queue until the transaction has been committed, and that's what we should do here too.

**Comment by diox** [comment]
I ended up fixing the main issues spotted by QA (https://github.com/mozilla/addons-server/issues/21996#issuecomment-2058720088) as well as the trademark ones in 2 different follow-ups:
- https://github.com/mozilla/addons-server/pull/22150
- https://github.com/mozilla/addons-server/pull/22154

I don't think we care about the rest as they shouldn't happen in production, but we'll see once the script is done re-running on dev and that we've tested stage. Worst case scenario we can address those in a new issue later.

**Comment by ioanarusiczki** [comment]
This has been verified on -dev and -stage, [this is the list](https://docs.google.com/document/d/1SnwnUYQGePKcZnl_2saPwj7tDqpJAD9h8punFHPSv4Y/edit?usp=sharing) of extensions that have been tested and includes extensions, themes and dictionaries with a last public version submitted before 2019-04-05.

Tested: 
- that theme previews are displayed correctly 
- installs of the old versions with Nightly 125.0a1
- installs of the new resigned versions with Nightly 127.0a1
- updates from a resigned version to a new submitted version on 127.0a1

**Comment by KevinMind** [comment]
Old Jira Ticket: https://mozilla-hub.atlassian.net/browse/ADDSRV-754
